### DISCUSSION TITLE
Incorrect filename

### DISCUSSION THREAD

**Discussion by mariusrak** [ImageMagick/ImageMagick#6270] [Help]
### ImageMagick version

7.1.1-6 Q16 x86

### Operating system

Windows

### Operating system, version and so on

22H2

### Description

I use command `convert` to resize images. In some cases, the output filename gets `-0` appended. It seems, that imagick adds ["filename reference"](https://imagemagick.org/script/command-line-processing.php#output). But the source is just an image, not PDF file. 

I noticed, this happens when I upload an example of malicious image from scenario #2 from https://dzone.com/articles/some-image-based-exploits-with-their-prevention 

But after I upload that image, it happens also with other non-malicious images uploaded.

Is it some sort of bug or am I doing something wrong?

### Steps to Reproduce

I use command `convert` to resize images. E.g.:

```
convert path/to/src.jpg -resize 100x100^ -gravity Center -crop 100x100^ path/to/src_100.jpg
```

In some cases, the output file is not `src_100.jpg`. but `src_100-0.jpg`.

I noticed, this happens when I upload an example of malicious image from scenario #2 from https://dzone.com/articles/some-image-based-exploits-with-their-prevention 

But after I upload that image, it happens also with other non-malicious images uploaded.

### Images

Image from scenario #2 from https://dzone.com/articles/some-image-based-exploits-with-their-prevention

**Comment by snibgo** [comment]
You use IM v7, so I suggest using `magick` so it uses v7 syntax, not `convert` or `magick convert` which both use the old v6 syntax.

> -crop 100x100

That's the problem. If you want only one image, use "`-crop 100x100+0+0`".

See https://imagemagick.org/script/command-line-options.php#crop :

> If the x and y offsets are present, a single image is generated, consisting of the pixels from the cropping region ...
> If the x and y offsets are omitted, a set of tiles of the specified geometry, covering the entire input image, is generated.
